components/ProductDisplay.js

```diff
@@ -0,0 +1,60 @@
+app.component('product-display',{
+    props:{
+        premium:{
+            type:Boolean,
+            required:true
+        }
+    },
+    template: 
+    /*html*/
+    `<div class="product-display">
+        <div class="product-container">
+          <product-image :image="image"></product-image>
+          <div class="product-info">
+           <product-title :title="title"></product-title>
+            <stock-availability :availability="inStock"/>
+            <shipping-info :shipping="shipping"/>
+            <product-details :details="details"/>
+            <product-variant-list :list="variants" :onClick="updateVariant"/>
+            <button class="button" :class="{ disabledButton: !inStock }" :disabled="!inStock" v-on:click="addToCart">Add to Cart</button>
```

Comment:
Button could also be its own component. please refactor accordingly

Reply:
no it can't